build/apple/download_swift_format.py

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env vpython3
+
+# Copyright (c) 2024 The Brave Authors. All rights reserved.
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import argparse
+import os
+import hashlib
+import deps
+
+from brave_chromium_utils import wspath
+from deps_config import DEPS_PACKAGES_URL
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Download swift-format')
+    parser.add_argument('version')
+    parser.add_argument('sha256hash')
+    args = parser.parse_args()
+
+    url = f'{DEPS_PACKAGES_URL}/swift-format/swift-format-{args.version}.zip'
+    # Overwrite Chromium's swift-format version as it very out of date and
+    # we will not be editing Chromium swift files to format anyways.
+    output_dir = wspath('//third_party/swift-format')
+    swift_format_bin = os.path.join(output_dir, 'swift-format')
+    if get_swift_format_hash(swift_format_bin) != args.sha256hash:
+        deps.DownloadAndUnpack(url, output_dir)
+        # Add execute privileges since they are not preserved in the zip
+        os.chmod(swift_format_bin, 0o755)
+
+
+def get_swift_format_hash(swift_format_bin):
+    with open(swift_format_bin, 'rb') as f:
+        return hashlib.sha256(f.read()).hexdigest()
```

Comment:
f.read() will read everything into memory. Let's use the efficient digest here: https://docs.python.org/3/library/hashlib.html#hashlib.file_digest